### PR TITLE
Exclude users from branch protection

### LINKED ISSUE #153: user config to allow overrid of Branch Protection
#### Problem to be solved

The automatic release build of a project is for us always done on the main branch. But this doesn't work if the Branch Protection is active for this branch (so that the main branch can only be changed with Pull Requests).
This prevents the build process from automatically changing the version number for the release of a new version.

#### Proposed solution

It would be good to have a config option that allows to define a user that can make direct changes on the protected branches to allow for automated tools to make the above described changes. 
Maybe with the restriction that one or more paths must be definded to the files that should be allowed to be changed.

### PR DESCRIPTION
**Pull request by pfeuffer**
## Proposed changes

Fixes #153 

This adds the option, to exclude users from the branch protection, enabling them to directly change otherwise protected
branches without the need for pull requests.

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] New code is covered with unit tests
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [x] Feature has been tested with different permissions
- [x] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+### Added
+- Exclude users from branch protection ([#167](https://github.com/scm-manager/scm-review-plugin/pull/167))
+
```
Comment (eheimbuch) [inline CHANGELOG.md:10]: This is okay, but we should start creating changelog entry files in all plugins.

File: src/main/java/com/cloudogu/scm/review/config/api/PullRequestConfigDto.java [hunk 2]
```diff
@@ -41,4 +41,11 @@ public PullRequestConfigDto(Links links) {
   private boolean preventMergeFromAuthor;
   private boolean restrictBranchWriteAccess;
   private List<String> protectedBranchPatterns;
+  private List<ExceptionEntryDto> protectedBranchExceptions;
+
+  @Getter @Setter
+  public static class ExceptionEntryDto {
```
Comment (eheimbuch) [inline src/main/java/com/cloudogu/scm/review/config/api/PullRequestConfigDto.java:47]: I was confused for a second why this is named exception. I can't think of a better name, though.

File: src/main/java/com/cloudogu/scm/review/config/service/ConfigService.java [hunk 3]
```diff
@@ -83,14 +88,31 @@ public boolean isPreventMergeFromAuthor(Repository repository) {
...
+      return getProtectedBranches(getGlobalPullRequestConfig());
     } else {
       return emptyList();
     }
   }
 
+  private Collection<String> getProtectedBranches(PullRequestConfig config) {
+    String user = SecurityUtils.getSubject().getPrincipal().toString();
+    Set<String> groups = groupCollector.collect(user);
+    if (config.getProtectedBranchExceptions().stream().anyMatch(exceptionEntry -> exceptionMatchesUser(exceptionEntry, user, groups))) {
```
Comment (eheimbuch) [inline src/main/java/com/cloudogu/scm/review/config/service/ConfigService.java:102]: Wrap this check into a new method. Just for the readability.

**Review by eheimbuch — CHANGES_REQUESTED** [review]

File: src/main/java/com/cloudogu/scm/review/config/api/PullRequestConfigDto.java (hunk 2 above)
Comment (pfeuffer) [reply src/main/java/com/cloudogu/scm/review/config/api/PullRequestConfigDto.java:47]: `ProtectionBypass` or simply `bypass` it will be...

**Pull request merged**